pkg/apis/core/v1alpha1/clusteroperation.go

```diff
@@ -0,0 +1,91 @@
+package v1alpha1
+
+import (
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ClusterOperationResourceName is name for "clusteroperation" resource
+	ClusterOperationResourceName = "clusteroperation"
+	// ClusterOperationResourcePlural is plural for "clusteroperation" resource
+	ClusterOperationResourcePlural = "clusteroperations"
+	// ClusterOperationShortName  is the shortname for "clusteroperation" resource
+	ClusterOperationShortName = "cor"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:deepcopy-gen=true
+// ClusterOperation represents a ClusterOperation resource
+type ClusterOperation struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata"`
+	Spec               ClusterOperationSpec   `json:"spec"`
+	Status             ClusterOperationStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterOperationList is a list of ClusterOperation
+type ClusterOperationList struct {
+	meta_v1.TypeMeta `json:",inline"`
+	meta_v1.ListMeta `json:"metadata"`
+	Items            []ClusterOperation `json:"items"`
+}
+
+// ClusterOperationSpec is the spec used to define a ClusterOperation
+type ClusterOperationSpec struct {
+	PoolSelector ClusterOperationPoolSelector `json:"poolSelector"`
```

Comment:
We can't have the pool selector in the generic ClusterOperationSpec struct since this is going to be used for non-pool cluster operations.